### PR TITLE
Refactor root resource to spec-first

### PR DESCRIPTION
**Pull request by kahowell**
Until #2548 is merged, this is just a draft, but can serve as an example to help understand the process outlined in that PR.

### CODE REVIEW

File: api/candlepin-api-spec.yaml [hunk 1]
```diff
@@ -10,7 +10,29 @@ servers:
...
 
-paths: {}
+paths:
+  /:
+    get:
+      description: Retrieves a list of links corresponding to Root resources
+      tags:
+        - root
+      operationId: getRootResources
+      security: []  # this resource doesn't require auth
```
Comment (Ceiu) [inline api/candlepin-api-spec.yaml]: This is probably a bit late to be asking this, but what is the intent for security here? Do we just throw our existing verify annotations in here for now (where appropriate) and then change this when we get around to updating our security model?

File: api/candlepin-api-spec.yaml [hunk 2]
```diff
@@ -22,6 +44,14 @@ components:
         requestUuid:
           type: string
 
+    Link:
+      description: Used to just represent a link to a top level resource in Candlepin
```
Comment (Ceiu) [inline api/candlepin-api-spec.yaml]: Grammar police: remove "just"

File: api/candlepin-api-spec.yaml (hunk 1 above)
Comment (kahowell) [reply api/candlepin-api-spec.yaml]: Yes, I think we can rework our security model in the future. Going ahead and specifying auth or noauth (by adding the line above) conveys intent in the spec, but we should keep our custom security annotations on the resource impl for now.

**Comment by kahowell** [comment]
23ca10734..6fb4f9084: grammar fix

Note this needs #2575 before tests will pass.

**Comment by nikosmoum** [comment]
Now that #2575 has been merged, this can be rebased right?

**Pull request merged**